packages/powersync_core/lib/src/streaming_sync.dart

```diff
@@ -244,7 +247,11 @@ class StreamingSyncImplementation implements StreamingSync {
           _updateStatus(uploading: false);
         }
       }
-    }, timeout: retryDelay);
+    }, timeout: retryDelay).whenComplete(() {
+      assert(identical(_activeCrudUpload, completer));
```

Comment:
For my understanding: Under what circumstances would this assertion fail and how would we recover from this?

Reply:
Hopefully, in no circumstances at all. We assert that there aren't two concurrent calls to `_uploadAllCrud()` (which is aready ensured by our code because `_uploadAllCrud()` is only called by `crudLoop`, which in turn is only called once in `streamingSync()`). I just wanted to spell that assumption out in code.